velox/duckdb/conversion/DuckConversion.cpp

```diff
@@ -48,6 +49,19 @@ variant decimalVariant(const Value& val) {
     case ::duckdb::PhysicalType::INT64: {
       return variant(val.GetValueUnsafe<int64_t>());
     }
+    case ::duckdb::PhysicalType::VARCHAR: {
```

Comment:
I'm wondering why are we expecting Decimal value in DuckDB to be represented using VARCHAR?

Reply:
I see. Looks like this test code assumes partition key is a string. I imagine it needs to change to allow for other types (possibly by adding cast to DuckDB SQL).

Reply:
Makes sense. Updated this PR to use cast in DuckDB SQL.